js/src/Contexts/WidgetManager.js

```diff
@@ -102,6 +95,25 @@ export class WidgetManager extends Component {
     this.componentDidUpdate();
   }
 
+  getCookie() {
```

Comment:
This seems like a really hard way to do this.  It looks like you're decoding the cookie by parsing it character by character.  Surely there must be some cookie library or something that we could use that would be a more efficient solution?

Reply:
well there's this https://www.npmjs.com/package/react-cookie but that adds some unnecessary weight for this task.